File: frontend/src/app/services/facility/facility.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environment/environment.prod';
import { Facility, FacilityDTO } from '../../models/facility.model';

@Injectable({
  providedIn: 'root'
})
export class FacilityService {
  private apiUrl = `${environment.apiUrl}/facility`;

  constructor(private http: HttpClient) { }

  createFacility(facility: FacilityDTO): Observable<FacilityDTO> {
    return this.http.post<FacilityDTO>(this.apiUrl, facility);
  }

  updateFacility(id: string, facility: FacilityDTO): Observable<FacilityDTO> {
    return this.http.put<FacilityDTO>(`${this.apiUrl}/${id}`, facility);
  }

  deleteFacility(id: string): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }

  getFacilitiesByTourId(tourId: string): Observable<FacilityDTO[]> {
    return this.http.get<FacilityDTO[]>(`${this.apiUrl}/tour/${tourId}`);
  }
}

File: backend/tour/src/main/java/com/group4/tour/service/FacilityService.java
package com.group4.tour.service;

import com.group4.tour.data.model.Facility;
import com.group4.tour.dto.FacilityDTO;

import java.util.List;

public interface FacilityService {
    Facility createFacility(FacilityDTO facilityDTO);
    FacilityDTO updateFacility(String id, FacilityDTO facilityDTO);
    void deleteFacility(String id);
    List<Facility> getFacilitiesByTourId(String tourId);
}